File: Lista de Exercicio Paradigmas/Lista Java/Ex6.java
class Motor {
    private int potencia;
    private String tipo;

    public Motor(int potencia, String tipo) {
        this.potencia = potencia;
        this.tipo = tipo;
    }

    public String detalhes() {
        return "Motor: " + this.potencia + " CV, Tipo: " + this.tipo;
    }
}

class Carro {
    private String marca;
    private String modelo;
    private int ano;
    private Motor motor;

    public Carro(String marca, String modelo, int ano, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.motor = motor;
    }

    public String detalhes() {
        return "Carro: " + this.marca + " " + this.modelo + ", Ano: " + this.ano + ", " + this.motor.detalhes();
    }

    public static void main(String[] args) {
        Motor motor1 = new Motor(150, "Gasolina");
        Carro carro1 = new Carro("Toyota", "Corolla", 2020, motor1);

        Motor motor2 = new Motor(120, "Etanol");
        Carro carro2 = new Carro("Honda", "Civic", 2019, motor2);

        System.out.println(carro1.detalhes());
        System.out.println(carro2.detalhes());
    }
}

File: Lista de Exercicio Paradigmas/Lista Python/Ex6.py
class Motor:
    def __init__(self, potencia, tipo):
        self.potencia = potencia  
        self.tipo = tipo  

    def detalhes(self):
        return f'Motor: {self.potencia} CV, Tipo: {self.tipo}'

class Carro:
    def __init__(self, marca, modelo, ano, motor):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.motor = motor  

    def detalhes(self):
        return f'Carro: {self.marca} {self.modelo}, Ano: {self.ano}, {self.motor.detalhes()}'

# Exemplo de uso
motor1 = Motor(150, 'Gasolina')
carro1 = Carro('Toyota', 'Corolla', 2020, motor1)

motor2 = Motor(120, 'Etanol')
carro2 = Carro('Honda', 'Civic', 2019, motor2)

print(carro1.detalhes())
print(carro2.detalhes())